### PR TITLE
Feature missions

### PR DESCRIPTION
**Pull request by planes**

### CODE REVIEW

File: trunk/SUAVE/Analyses/Mission/Segments/Climb/Constant_Speed_Linear_Altitude.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
+# Constant_Speed_Constant_Angle.py
```
Comment (timdmacdo) [inline trunk/SUAVE/Analyses/Mission/Segments/Climb/Constant_Speed_Linear_Altitude.py]: wrong name

File: trunk/SUAVE/Analyses/Mission/Segments/Climb/Unknown_Throttle.py [hunk 2]
```diff
@@ -43,7 +44,7 @@ def __defaults__(self):
         # initials and unknowns
         ones_row = self.state.ones_row
         self.state.unknowns.throttle   = ones_row(1) * 0.5
-        self.state.unknowns.body_angle = ones_row(1) * 0.0
+        self.state.unknowns.body_angle = ones_row(1) * 0.01
```
Comment (timdmacdo) [inline trunk/SUAVE/Analyses/Mission/Segments/Climb/Unknown_Throttle.py]: What's the reason for this change?

File: trunk/SUAVE/Analyses/Mission/Segments/Cruise/Constant_Speed_Constant_Altitude.py [hunk 3]
```diff
@@ -45,7 +45,7 @@ def __defaults__(self):
         # initials and unknowns
         ones_row = self.state.ones_row
         self.state.unknowns.throttle   = ones_row(1) * 0.5
-        self.state.unknowns.body_angle = ones_row(1) * 0.0
+        self.state.unknowns.body_angle = ones_row(1) * 1.0 * Units.deg
```
Comment (timdmacdo) [inline trunk/SUAVE/Analyses/Mission/Segments/Cruise/Constant_Speed_Constant_Altitude.py]: and also this change?

File: trunk/SUAVE/Methods/Missions/Segments/Climb/Constant_Dynamic_Pressure_Constant_Angle.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    state.conditions.propulsion.throttle[:,0]            = throttle[:,0]
+    state.conditions.frames.body.inertial_rotations[:,1] = theta[:,0]  
+    
+
+def residual_total_forces(segment,state):
+    
+    FT = state.conditions.frames.inertial.total_force_vector
+    a  = state.conditions.frames.inertial.acceleration_vector
+    m  = state.conditions.weights.total_mass    
+    alt_in  = state.unknowns.altitudes[:,0]
```
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Missions/Segments/Climb/Constant_Dynamic_Pressure_Constant_Angle.py]: It's not very clear to me what's happening here. Could use comments. Also this doesn't seem like it belong in residuals.forces

File: trunk/SUAVE/Methods/Missions/Segments/Climb/Constant_Mach_Constant_Angle.py [hunk 5]
```diff
@@ -43,5 +49,20 @@ def initialize_conditions(segment,state):
...
+    conditions.frames.inertial.position_vector[:,2] = -alts[:,0] # z points down
+    state.conditions.propulsion.throttle[:,0]            = throttle[:,0]
+    state.conditions.frames.body.inertial_rotations[:,1] = theta[:,0]  
+    
+def residual_total_forces(segment,state):
+    
+    FT = state.conditions.frames.inertial.total_force_vector
+    a  = state.conditions.frames.inertial.acceleration_vector
+    m  = state.conditions.weights.total_mass    
+    alt_in  = state.unknowns.altitudes[:,0]
```
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Missions/Segments/Climb/Constant_Mach_Constant_Angle.py]: same comments as earlier

File: trunk/SUAVE/Methods/Missions/Segments/Climb/Optimized.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+def initialize_unknowns(segment,state):
+    
+    # unpack unknowns and givens
+    gamma    = state.unknowns.flight_path_angle
+    vel      = state.unknowns.velocity 
+    v0       = segment.air_speed_start
+    vf       = segment.air_speed_end 
+    ones     = state.ones_row(1)
+    ones_m1  = state.ones_row_m1(1)
```
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Missions/Segments/Climb/Optimized.py]: Might be useful to note what these are.

File: trunk/SUAVE/Methods/Missions/Segments/Climb/Optimized.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    conditions.frames.inertial.time[:,0]            = t_initial + x[:,0] 
+    conditions.frames.inertial.position_vector[:,2] = -alt[:,0] # z points down
+    conditions.freestream.altitude[:,0]             =  alt[:,0] # positive altitude in this context    
+
+    return
+
+def objective(segment,state):
+    
+    if segment.objective is not None:
+        if segment.minimize ==True:
```
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Missions/Segments/Climb/Optimized.py]: I don't understand the behavior here, but that should be fine as long as it's documented later. Also you missed a space

File: trunk/SUAVE/Methods/Missions/Segments/optimize.py [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    if not np.all(state.inputs_last == state.unknowns):       
+        segment.process.iterate(segment,state)
+    
+    # Time goes forward, not backward
+    t_final = state.conditions.frames.inertial.time[-1,0]
+    constraints = (state.conditions.frames.inertial.time[1:,0] - state.conditions.frames.inertial.time[0:-1,0])/t_final
+    
+    # Less than a specified CL limit
+    CL_limit = segment.CL_limit 
+    constraints2 = (CL_limit  - state.conditions.aerodynamics.lift_coefficient[:,0])/CL_limit
```
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Missions/Segments/optimize.py]: Use more descriptive names here

File: trunk/SUAVE/Analyses/Mission/Segments/Climb/Constant_Speed_Linear_Altitude.py (hunk 1 above)
Comment (planes) [inline trunk/SUAVE/Analyses/Mission/Segments/Climb/Constant_Speed_Linear_Altitude.py]: My branch has this name less..

File: trunk/SUAVE/Analyses/Mission/Segments/Climb/Unknown_Throttle.py (hunk 2 above)
Comment (planes) [inline trunk/SUAVE/Analyses/Mission/Segments/Climb/Unknown_Throttle.py]: To make it faster to converge and tell what this is. Changed to degrees.

File: trunk/SUAVE/Analyses/Mission/Segments/Climb/Constant_Speed_Linear_Altitude.py (hunk 1 above)
Comment (planes) [inline trunk/SUAVE/Analyses/Mission/Segments/Climb/Constant_Speed_Linear_Altitude.py]: Not found on local branch. Updated regardless.

File: trunk/SUAVE/Analyses/Mission/Segments/Cruise/Constant_Speed_Constant_Altitude.py (hunk 3 above)
Comment (planes) [inline trunk/SUAVE/Analyses/Mission/Segments/Cruise/Constant_Speed_Constant_Altitude.py]: This is to tell what angle the vehicle is at for debugging when unpacking an packing. Changed to match prior one.

File: trunk/SUAVE/Methods/Missions/Segments/Climb/Constant_Dynamic_Pressure_Constant_Angle.py (hunk 4 above)
Comment (planes) [inline trunk/SUAVE/Methods/Missions/Segments/Climb/Constant_Dynamic_Pressure_Constant_Angle.py]: Added.

File: trunk/SUAVE/Methods/Missions/Segments/Climb/Optimized.py (hunk 6 above)
Comment (planes) [inline trunk/SUAVE/Methods/Missions/Segments/Climb/Optimized.py]: Commented

File: trunk/SUAVE/Methods/Missions/Segments/optimize.py (hunk 8 above)
Comment (planes) [inline trunk/SUAVE/Methods/Missions/Segments/optimize.py]: Names changed.

**Comment by planes** [comment]
A regression would be good for all mission to be solved sequentially. We could remove B737. Will discuss in weekly meetings.

File: trunk/SUAVE/Methods/Missions/Segments/Climb/Linear_Speed_Constant_Rate.py [hunk 9]
```diff
@@ -40,7 +39,7 @@ def initialize_conditions(segment,state):
     v_x   = np.sqrt( v_mag**2 - v_z**2 )
     
     # pack conditions    
-    conditions.frames.inertial.velocity_vector[:,0] = v_x
+    conditions.frames.inertial.velocity_vector[:,0] = v_x[:,0]
```
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Missions/Segments/Climb/Linear_Speed_Constant_Rate.py]: Have you checked that v_x[:,0] and v_z (without the brackets) give the same format?

File: trunk/SUAVE/Analyses/Mission/Segments/Climb/Optimized.py [hunk 10]
```diff
@@ -64,9 +66,9 @@ def __defaults__(self):
         #   Initialize - before iteration
         # --------------------------------------------------------------
         initialize = self.process.initialize
-        
         initialize.expand_state            = Methods.expand_state
-        initialize.unknowns                = Methods.Climb.Optimized.initialize_unknowns
+        initialize.solved_mission          = Methods.Climb.Optimized.solve_linear_speed_constant_rate
+        #initialize.unknowns                = Methods.Climb.Optimized.initialize_unknowns
```
Comment (timdmacdo) [inline trunk/SUAVE/Analyses/Mission/Segments/Climb/Optimized.py]: It looks like you deleted this function later, so maybe remove this comment as well.

File: trunk/SUAVE/Methods/Missions/Segments/Climb/Linear_Speed_Constant_Rate.py (hunk 9 above)
Comment (planes) [inline trunk/SUAVE/Methods/Missions/Segments/Climb/Linear_Speed_Constant_Rate.py]: Yes. v_x is a vector now. It was a scalar before... v_z is a scalar because it is constant rate. There was an error here prior because it was copy of a CSCR segment, which has v_x and v_z as both scalars.

File: trunk/SUAVE/Analyses/Mission/Segments/Climb/Optimized.py (hunk 10 above)
Comment (planes) [inline trunk/SUAVE/Analyses/Mission/Segments/Climb/Optimized.py]: Good catch.

**Comment by mclarke2** [comment]
Confirmation that I reviewed this pull request. All of the changes look good

**Comment by timdmacdo** [comment]
Great let's pull it in
